include/sentry.h

```diff
@@ -1226,6 +1226,50 @@ SENTRY_EXPERIMENTAL_API void sentry_options_set_traces_sample_rate(
 SENTRY_EXPERIMENTAL_API double sentry_options_get_traces_sample_rate(
     sentry_options_t *opts);
 
+/* -- Performance Monitoring/Tracing APIs -- */
+
+/**
+ * Constructs a new transaction context to be passed
+ * into `sentry_start_transaction`.
+ */
+SENTRY_EXPERIMENTAL_API sentry_value_t sentry_value_new_transaction_context(
+    const char *name);
+
+/**
+ * Sets the name of a transaction on a `transaction_context`.
+ * Returns 1 if the name provided is invalid (e.g. empty or nil).
```

Comment:
Is this consistent with other similar APIs? I don’t quite remember if we even validate invalid values?

Reply:
oh yeah, good catch. [python](https://github.com/getsentry/sentry-python/blob/dd0efc08414ee2ef1a5f22d2cc4e243b54a1b455/sentry_sdk/tracing.py#L565-L568) and [js](https://github.com/getsentry/sentry-javascript/blob/5bba9b44b47ffef3c199ed6361df046009b789c4/packages/tracing/src/transaction.ts#L95-L96) merely warn and set the value to `<unlabeled transaction>`, so i'll fall back to that.

Reply:
When I was experimenting with this in Rust, the server (relay?) would also default this. So the SDK is free to leave off the name completely (or make it empty) instead of defaulting it.